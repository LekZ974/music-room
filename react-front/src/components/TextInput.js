import React from 'react';

import { Input as NBInput, Item } from 'native-base';
import PropTypes from 'prop-types';

export default function TextInput(props) {
  const { input, ...inputProps } = props;

  return (
    <Item>
      <NBInput
        {...inputProps}
        onChangeText={input.onChange}
        onBlur={input.onBlur}
        onFocus={input.onFocus}
        value={input.value}
      />
    </Item>
  );
}

TextInput.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    onBlur: PropTypes.func,
    onFocus: PropTypes.func,
    value: PropTypes.string,
  }).isRequired,
};
