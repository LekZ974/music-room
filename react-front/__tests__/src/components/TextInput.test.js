import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from '../../../src/components/TextInput';

it('renders correctly', () => {
  const tree = renderer
    .create(<TextInput input={{ placeholder: 'Snapshot test with TextInput Component!' }} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
