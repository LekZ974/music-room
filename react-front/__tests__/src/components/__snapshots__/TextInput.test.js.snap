// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomWidth": 1,
      "borderColor": "#D9D5DC",
      "borderLeftWidth": 0,
      "borderRadius": undefined,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "borderWidth": undefined,
      "flexDirection": "row",
      "marginLeft": 2,
      "opacity": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    editable={true}
    placeholderTextColor="#575757"
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontSize": 17,
        "height": 50,
        "paddingLeft": 5,
        "paddingRight": 5,
        "top": 1.5,
      }
    }
    underlineColorAndroid="rgba(0,0,0,0)"
  />
</View>
`;
